assets/js/components/higherorder/withData.js

```diff
@@ -153,100 +176,141 @@ const withData = (
 				moduleRequiringSetup: '',
 			};
 
+			this.handleModuleDataReset = this.handleModuleDataReset.bind( this );
+			this.handleReturnedData = this.handleReturnedData.bind( this );
+			this.addDataRequests = this.addDataRequests.bind( this );
+			this.removeDataRequests = this.removeDataRequests.bind( this );
```

Comment:
The main reason for these changes is that initially I had simply updated the filter callback to apply the global date range as a default. This worked for the initial render, but because the callbacks were hooked in once from the component's constructor, they didn't respond to date range changes even when the prop was being passed through correctly.

The component also does a better job of cleaning up after itself now; where it previously would have caused potential memory leaks due to its callbacks never being removed.